airbyte-integrations/connectors/source-slack/source_slack/spec.json

```diff
@@ -0,0 +1,144 @@
+{
+  "documentationUrl": "https://docs.airbyte.com/integrations/sources/slack",
+  "connectionSpecification": {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Slack Spec",
+    "type": "object",
+    "required": ["start_date", "lookback_window", "join_channels"],
+    "additionalProperties": true,
+    "properties": {
+      "start_date": {
+        "type": "string",
+        "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$",
+        "description": "UTC date and time in the format 2017-01-25T00:00:00Z. Any data before this date will not be replicated.",
+        "examples": ["2017-01-25T00:00:00Z"],
+        "title": "Start Date"
+      },
+      "lookback_window": {
+        "type": "integer",
+        "title": "Threads Lookback window (Days)",
+        "description": "How far into the past to look for messages in threads.",
+        "examples": [7, 14]
+      },
+      "join_channels": {
+        "type": "boolean",
+        "default": true,
+        "title": "Join all channels",
+        "description": "Whether to join all channels or to sync data only from channels the bot is already in.  If false, you'll need to manually add the bot to all the channels from which you'd like to sync messages. "
+      },
+      "channel_filter": {
+        "type": "array",
+        "default": [],
+        "items": {
+          "type": "string",
+          "minLength": 0
+        },
+        "title": "Channel name filter",
+        "description": "A channel name list (without leading '#' char) which limit the channels from which you'd like to sync. Empty list means no filter.",
+        "examples": ["channel_one", "channel_two"]
+      },
+      "credentials": {
+        "title": "Authentication mechanism",
+        "description": "Choose how to authenticate into Slack",
+        "type": "object",
+        "oneOf": [
```

Comment:
When we have a `oneOf`, it's best to provide a `discriminator` property which specifies the field that discriminates between the two options most clearly. In this case, the `option_title` identifies OAuth vs API Token credentials.

So it would be something like this:
```
discriminator: {
  "propertyName": "option_title"
}
```